test/dynamo/test_aot_autograd_cache.py

```diff
@@ -105,6 +110,45 @@ def fn(x):
         c2 = self.gen_cache_key(fn, config2)
         self.assertNotEqual(c1, c2)
 
+    def test_different_inputs(self):
+        def fn(x):
+            return x.cos().sin()
+
+        config = self.default_config()
+        c1 = self.gen_cache_key(fn, config, inputs=[torch.ones(3)])
+        c2 = self.gen_cache_key(fn, config, inputs=[torch.ones(2)])
+        self.assertNotEqual(c1, c2)
+
+    def test_different_global_configs(self):
+        def fn(x):
+            return x.cos().sin()
+
+        config = self.default_config()
+
+        c1 = self.gen_cache_key(fn, config)
+        c2 = self.gen_cache_key(fn, config)
+        self.assertEqual(c1, c2)
+
+        c1 = self.gen_cache_key(fn, config)
+        # Change global config
+        functorch_config.debug_assert = True
```

Comment:
a) Should you save/restore the old value so that if we run this test in the middle of a bunch of other tests it doesn't corrupt tests after it? (I don't know if the test harness resets the global configs between each test or not) (Same with the config changes below)

b) What if `debug_assert` was already True when this test started? Maybe 
```
functorch_config.debug_assert = not functorch_config.debug_assert
```
(Same with the config changes below)

Reply:
Good point, I can use config.patch to do this a bit better